### DISCUSSION TITLE
Why the color of bars changes each time I draw a plot with "colour.plotting.plot_single_sd_colour_rendering_index_bars" definition?

### DISCUSSION THREAD

**Discussion by amons000** [colour-science/colour#723] [Q&A] [answered]
I use `colour.plotting.plot_single_sd_colour_rendering_index_bars` to draw a bar diagram.
I suppose each time each bar should have the same color so I could compare them more intuitively.
but the bars showing in different color sets.

---
![1](https://user-images.githubusercontent.com/35733429/57191517-4ad8b780-6f59-11e9-9539-3d14436930da.png)

---
![2](https://user-images.githubusercontent.com/35733429/57191522-59bf6a00-6f59-11e9-93d3-e61792603f19.png)

**Comment by thomasmansencal** [comment, marked as answer]
Hi @Edwardlin-zlt,

A typical plot would look like this:

![image](https://user-images.githubusercontent.com/99779/57191622-cea39b80-6f7b-11e9-8b20-4bc91080632d.png)

The bars are actually coloured the way the light tested would render the test sample.

Are your lights very narrowband/highly colourful?

**Comment by amons000** [comment]
Thanks~

 My lights are not real lights. And it has low rf.

What if I really want those bar's have one exactly same colour set? What I could do to reach that goal?

And is it normal to have one bar above 100 line? You see my outcome plot has a bar out of figure box.

**Comment by thomasmansencal** [comment]
> What if I really want those bar's have one exactly same colour set? What I could do to reach that goal?

I'm a bit unclear, do you want them to have all the same colours or use constant pre-defined colours?

> And is it normal to have one bar above 100 line? You see my outcome plot has a bar out of figure box.

I would say yes, CRI was fitted against real-world lights with relatively broad spectrum so to speak, if you use narrowband artificial lights I would not be surprised that you get such results, it is also possible to get negative values, although those are taken in account and all the bars will be showing as positive.

**Comment by amons000** [comment]
Sorry for the late. For the first one, I actually want control the bars' colour.

So what could I do if I don't want bars to be coloured the way the light tested would render the test sample, just want them to be the same colour each time i draw the plot.

**Comment by thomasmansencal** [comment]
Something like that should work:

```python
import colour
import colour.plotting

illuminant = colour.ILLUMINANTS_SDS['FL2']
_figure, axes = colour.plotting.plot_single_sd_colour_quality_scale_bars(
    illuminant, standalone=False)

for patch in axes.patches:
    patch.set_facecolor('r')

colour.plotting.render(standalone=True)
```

**Comment by amons000** [comment]
Thanks！

You are so patient and helpful~~~~
